metrics/utils/preprocess.py

```diff
@@ -0,0 +1,64 @@
+import os
+import ffmpeg
+import subprocess
+
+def extract_frames(video_path, output_dir, width=1920, height=1080, fps=30, israwvideo=False):
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)
+    try:
+        if israwvideo:
+            (
+                ffmpeg
+                .input(video_path, s='{}x{}'.format(width, height), pix_fmt='yuv420p', r=fps)
+                .output(os.path.join(output_dir, '%04d.png'), start_number=0)
+                .run()
+            )
+        else:
+            (
+                ffmpeg
+                .input(video_path)
+                .output(os.path.join(output_dir, '%04d.png'), r=fps, start_number=0)
+                .run()
+            )
+    except ffmpeg.Error as e:
+        print(e.stderr)
+        raise e
+    
+def recognize_frame_numbers(frame_dir):
+    cut_folder = os.path.join(frame_dir, 'cut')
+    if not os.path.exists(cut_folder):
+        os.makedirs(cut_folder)
+
+    frames = sorted([f for f in os.listdir(frame_dir) if f.endswith('.png')])
+    for frame in frames:
+        frame_path = os.path.join(frame_dir, frame)
+        cut_frame_path = os.path.join(cut_folder, f'cut_{frame}')
+        
+        # Crop the frame number area
+        crop_value = '70x30+910+1030'  # Adjust this value based on your video resolution
```

Comment:
if we have changed the video resolution, how can we adjust it? Can we change it into a function and generate its value by uthe resolution getting from the input frame automatically?